spec/rubocop/cop/style/safe_navigation_spec.rb

```diff
@@ -933,6 +933,23 @@
           expect(new_source).to eq("#{variable}&.bar(baz) { |e| e.qux }\n")
         end
 
+        it 'does not lose comments within if expression' do
+          new_source = autocorrect_source(<<~RUBY)
+            if #{variable}
+              # this is a comment
+              # another comment
+              #{variable}.bar
+            end
+          RUBY
+
+          expected_source = <<~RUBY
+            # this is a comment
+            # another comment
+            #{variable}&.bar
+          RUBY
+          expect(new_source).to eq(expected_source)
+        end
```

Comment:
Hmm, indent seems to be broken when used in a method.

```diff
diff --git a/spec/rubocop/cop/style/safe_navigation_spec.rb b/spec/rubocop/cop/style/safe_navigation_spec.rb
index 2c142df19..10697d908 100644
--- a/spec/rubocop/cop/style/safe_navigation_spec.rb
+++ b/spec/rubocop/cop/style/safe_navigation_spec.rb
@@ -935,17 +935,21 @@ RSpec.describe RuboCop::Cop::Style::SafeNavigation, :config do

         it 'does not lose comments within if expression' do
           new_source = autocorrect_source(<<~RUBY)
-            if #{variable}
-              # this is a comment
-              # another comment
-              #{variable}.bar
+            def foo
+              if #{variable}
+                # this is a comment
+                # another comment
+                #{variable}.bar
+              end
             end
           RUBY

           expected_source = <<~RUBY
-            # this is a comment
-            # another comment
-            #{variable}&.bar
+            def foo
+              # this is a comment
+              # another comment
+              #{variable}&.bar
+            end
           RUBY
           expect(new_source).to eq(expected_source)
         end
```

```console
% bundle exec rspec spec/rubocop/cop/style/safe_navigation_spec.rb

(snip)

Failures:

  1) RuboCop::Cop::Style::SafeNavigation auto-correct behaves like all variable types if expression does not lose comments within if expression
     Failure/Error: expect(new_source).to eq(expected_source)

       expected: "def foo\n  # this is a comment\n  # another comment\n  FOO::BAR&.bar\nend\n"
            got: "def foo\n  # this is a comment\n# another comment\nFOO::BAR&.bar\nend\n"

       (compared using ==)

       Diff:
       @@ -1,6 +1,6 @@
        def foo
          # this is a comment
       -  # another comment
       -  FOO::BAR&.bar
       +# another comment
       +FOO::BAR&.bar
        end

     Shared Example Group: "all variable types" called from ./spec/rubocop/cop/style/safe_navigation_spec.rb:1193
     # ./spec/rubocop/cop/style/safe_navigation_spec.rb:954:in `block (5 levels) in <top (required)>'
```

Reply:
I think that's acceptable. Some of the `Layout` cops will fix the indentation in the auto-correct loop.

Reply:
I get it. @jonas054's opinion is excellent.